website_sale_require_legal/controllers/main.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingeniería, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class RequireLegalTermsToCheckout(website_sale):
+    def _get_mandatory_billing_fields(self, *args, **kwargs):
+        """Require accepting legal terms for unauthenticated users."""
```

Comment:
In Europe, personal data protection laws cover any person's data in any company. When you checkout, you are giving the company your personal information to buy (name, address, bank account, etc.). Any buyer should accept legal terms before that happens.